app/main.py

```diff
@@ -1,3 +1,12 @@
-def cache(func):
-    # Write your code here
-    pass
+def cache(func: callable) -> callable:
+    cache_dict = {}
+
+    def inner(*args: any) -> callable:
```

Comment:
incorrect typehint
function doesn't return a function and u don't need typehint for *args in this case